xwiki-rendering-legacy/xwiki-rendering-legacy-transformations/xwiki-rendering-legacy-transformation-macro/src/test/java/org/xwiki/rendering/macro/descriptor/DeprecatedAbstractMacroDescriptorTest.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.rendering.macro.descriptor;
+
+import java.util.Set;
+
+import org.junit.jupiter.api.Test;
+import org.xwiki.properties.BeanDescriptor;
+import org.xwiki.rendering.macro.MacroId;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.mock;
+
+/**
+ * Test of the deprecated methods of {@link AbstractMacroDescriptorAspect}.
+ *
+ * @version $Id$
+ * @since 14.6RC1
+ */
+class DeprecatedAbstractMacroDescriptorTest
+{
+    @Test
+    void setDefaultCategory()
+    {
+        AbstractMacroDescriptor abstractMacroDescriptor = new AbstractMacroDescriptor(new MacroId("testmacro"), "testmacro", "description",
+            new DefaultContentDescriptor("description", false), mock(BeanDescriptor.class))
+        {
+        };
+
+        abstractMacroDescriptor.setDefaultCategory("testcategory");
+        assertEquals(Set.of("testcategory"), abstractMacroDescriptor.getDefaultCategories());
```

Comment:
That looks like a possible flicker. You should use hamcrest API IMO.

Reply:
> Even if there were several, the order is not supposed to be taken into account when you compare two Set objects.

AFAIK there's no support for that in junit5's assertEquals. Unless you know for a fact that it's supported.

Reply:
Actually it seems you're right and that the Set's equals method support this OOB: https://docs.oracle.com/javase/1.5.0/docs/api/java/util/Set.html#equals%28java.lang.Object%29

Didn't know that.